packages/cashier/src/stores/account-transfer-store.js

```diff
@@ -482,8 +486,16 @@ export default class AccountTransferStore {
                     // check if selected to is not allowed account
                     obj_values.error = getSelectedError(obj_values.value);
                 }
-                // set the first available account as the default transfer to account
-                this.setSelectedTo(obj_values);
+
+                const { account_id, login } = this.root_store.traders_hub?.selected_account;
+
+                //if from appstore -> set selected account as the default transfer to account
+                //if not from appstore -> set the first available account as the default transfer to account
+                if (!is_from_pre_appstore) {
+                    this.setSelectedTo(obj_values);
+                } else if ([account_id, login].includes(account.loginid)) {
+                    this.setSelectedTo(obj_values);
```

Comment:
in if and else id you are following the same logic, should it be changed?

Reply:
@mahdiyeh-deriv oops yes, my fault, 🤦‍♂️, do refctoring